code/modules/projectiles/guns/ballistic.dm

```diff
@@ -257,6 +257,10 @@
 /obj/item/gun/ballistic/proc/rack(mob/user = null)
 	if (bolt_type == BOLT_TYPE_NO_BOLT) //If there's no bolt, nothing to rack
 		return
+	if (weapon_weight != WEAPON_LIGHT) //Can't rack it if the weapon doesn't permit dual-wielding and your off-hand is full
+		if (user.get_inactive_held_item())
+			to_chat(user, span_notice("You cannot rack the [bolt_wording] of \the [src] while your off-hand is full!"))
```

Comment:
```suggestion
	if (weapon_weight != WEAPON_LIGHT)
		if (user.get_inactive_held_item()) //Can't rack it if the weapon doesn't permit dual-wielding and your off-hand is full
			to_chat(user, span_warning("You cannot rack the [bolt_wording] of \the [src] while your off-hand is full!"))
```
i'd rather have it say other hand instead of off hand but you do you

Reply:
also does the text in the to chat come out right

Reply:
yes

Reply:
> i'd rather have it say other hand instead of off hand but you do you

yeah sure I'll get it done later probably